crates/moonbuild-rupes-recta/src/build_lower/compiler/link_core.rs

```diff
@@ -0,0 +1,251 @@
+        }
+        if self.flags.no_opt {
+            args.push("-O0".to_string());
+        }
+
+        // Source map
+        if self.flags.source_map {
+            args.push("-source-map".to_string());
+        }
+
+        // WASM-specific config
+        if matches!(
+            self.target_backend,
+            TargetBackend::Wasm | TargetBackend::WasmGC
+        ) {
+            // WebAssembly exports
+            if let Some(exports) = self.wasm_config.exports {
+                if exports.is_empty() {
+                    // Empty exports case - legacy adds empty string
+                    args.push("".to_string());
+                } else {
+                    args.push(format!("-exported_functions={}", exports.join(",")));
+                }
+            }
+
+            // Export memory name
+            if let Some(export_memory_name) = &self.wasm_config.export_memory_name {
+                args.push("-export-memory-name".to_string());
+                args.push(export_memory_name.to_string());
+            }
+
+            // Import memory
+            if let Some(import_memory) = self.wasm_config.import_memory {
+                args.push("-import-memory-module".to_string());
+                args.push(import_memory.module.clone());
+                args.push("-import-memory-name".to_string());
+                args.push(import_memory.name.clone());
+            }
+
+            // Memory limits
+            if let Some(memory_limits) = self.wasm_config.memory_limits {
+                args.push("-memory-limits-min".to_string());
+                args.push(memory_limits.min.to_string());
+                args.push("-memory-limits-max".to_string());
+                args.push(memory_limits.max.to_string());
+            }
+
+            // Shared memory
+            if let Some(shared_memory) = self.wasm_config.shared_memory {
+                if shared_memory {
+                    args.push("-shared-memory".to_string());
+                }
+            }
+
+            // Heap start address
+            if let Some(heap_start_address) = self.wasm_config.heap_start_address {
+                args.push("-heap-start-address".to_string());
+                args.push(heap_start_address.to_string());
+            }
+
+            // Link flags
+            if let Some(link_flags) = self.wasm_config.link_flags {
+                for flag in link_flags {
+                    args.push(flag.clone());
+                }
+            }
+        }
+
+        // JavaScript format (only for JS target)
+        if self.target_backend == TargetBackend::Js {
+            if let Some(js_format) = self.js_format {
+                args.push("-js-format".to_string());
+                args.push(js_format.to_flag().to_string());
+            } else {
+                // Use default JS format
+                args.push("-js-format".to_string());
+                args.push(JsFormat::default().to_flag().to_string());
```

Comment:
Nit: Collapse this `if` with `unwrap_or_default()`?

Reply:
This was written by AI, but good idea to fix it.